src/Illuminate/Database/Query/Builder.php

```diff
@@ -3158,7 +3158,7 @@ public function soleValue($column)
     {
         $result = (array) $this->sole([$column]);
 
-        return reset($result);
+        return array_last($result);
```

Comment:
Many in this file have had `reset` changed to `array_last` instead of `array_first`. 

Is it intentional?